framework.py

```diff
@@ -12,7 +12,12 @@
 import sys
 import os
 import struct
-import urlparse
+
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
```

Comment:
Could you use `six.moves.urllib` instead?
https://six.readthedocs.io/#module-six.moves.urllib.parse